src/util/task_runner.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_TASK_RUNNER_H
+#define BITCOIN_UTIL_TASK_RUNNER_H
+
+#include <cstddef>
+#include <functional>
+
+namespace util {
+
+/** @file
+ * This header provides an interface and simple implementation for a task
+ * runner. Another threaded, serial implementation using a queue is available in
+ * the scheduler module's SerialTaskRunner.
+ */
+
+class TaskRunnerInterface
+{
+public:
+    virtual ~TaskRunnerInterface() {}
+
+    /**
+     * The callback can either be queued for later/asynchronous/threaded
+     * processing, or be executed immediately for synchronous processing. When
+     * used in the ValidationSignals synchronous processing will block
+     * validation. Each subscriber will call this on each validation
+     * interface event.
+     */
+    virtual void insert(std::function<void()> func) = 0;
```

Comment:
What do you mean with this last sentence?
Validation events' subscribers will not call this function. This function will be called by the event dispatching side.